addons/resource/models/resource_calendar.py

```diff
@@ -193,6 +222,14 @@ def _get_hours_per_day(self, attendances):
 
         return float_round(hour_count / float(number_of_days), precision_digits=2)
 
+    def _get_days_per_week(self):
+        # If the employee didn't work a full day, it is still counted, i.e. 19h / week (M/T/W(half day)) -> 3 days
+        self.ensure_one()
+        days = len(set(self.attendance_ids.filtered(
+            lambda attendance_id: attendance_id._is_work_period() and attendance_id.duration_hours)
+            .mapped(lambda attendance_id: f"{attendance_id.week_type} {attendance_id.dayofweek}")))
```

Comment:
this line a bit too complex ^^'
```suggestion
        attendances = self.attendance_ids.filtered(
            lambda attendance_id: attendance_id._is_work_period() and attendance_id.duration_hours)
         attendances_set = attendances.mapped(lambda attendance: (attendance.week_type, attendance.dayofweek))
         number_of_days = len(attendances_set)  
        
```
And we write attendance_id only it's just the id not the recordset :)
What do you think ? is it easier for you to read or you think it's a bit too much ? :)